mypyc/genops.py

```diff
@@ -935,37 +937,49 @@ def visit_while_stmt(self, s: WhileStmt) -> Value:
         return INVALID_VALUE
 
     def visit_for_stmt(self, s: ForStmt) -> Value:
-        if (isinstance(s.expr, CallExpr)
-                and isinstance(s.expr.callee, RefExpr)
-                and s.expr.callee.fullname == 'builtins.range'):
+        def body() -> None:
+            self.accept(s.body)
+        return self.loop_helper(s.index, s.expr, body, s.line)
+
+    def loop_helper(self, index: Lvalue, expr: Expression,
+                    body_insts: Callable[[], None], line: int) -> Value:
+        '''
+        Generate IR for a loop.
+        "index" is the loop index Lvalue
+        "expr" is the expression to iterate over
+        "body_insts" is a function to generate the body of the loop.
+        '''
```

Comment:
Nit: the docstring format we use in mypy/mypyc is:
```
"""A one line summary or whatever.

Any additional lines after a blank line.
Stuff.
"""
```
(With double-quotes, apparently.)